wallet/src/Pos/Wallet/Web/Methods/Logic.hs

```diff
@@ -319,6 +320,14 @@ newExternalAccount
     -> m CAccount
 newExternalAccount = newExternalAccountIncludeUnready False
 
+doesWalletExist
+    :: MonadWalletLogic ctx m
+    => CId Wal -> m Bool
+doesWalletExist walletId = do
+    db <- askWalletDB
+    ws <- getWalletSnapshot db
+    return $ isJust $ getWalletMetaIncludeUnready ws True walletId
```

Comment:
This is a slippery-slope of a function -- you are clearly performing a query, but you are reading the DB every time! This would give inconsistent results if called twice (for example if in the middle you delete a wallet). For future references, think about writing such queries as `doesWalletExist :: WalletSnapshot -> CId Wal -> m Bool` 😉